src/D2L.Security.OAuth2.WebApi/Authorization/NoImpersonationAttribute.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Net;
+using System.Net.Http;
+using System.Web.Http.Controllers;
+using D2L.Security.OAuth2.Principal;
+
+namespace D2L.Security.OAuth2.Authorization {
+	/// <summary>
+	/// Block requests where the access token indicates user impersonation
+	/// </summary>
+	[AttributeUsage( AttributeTargets.All, AllowMultiple = false )]
+	public sealed class NoImpersonationAttribute : OAuth2AuthorizeAttribute {
+		protected override uint Order => 2;

```

Comment:
RE: https://ci.appveyor.com/project/Brightspace/d2l-security-oauth2/build/job/k83jy2420h5wustg/messages

.net 4.6.1?

FYI auth service is still at 4.5 (not sure if it matters).